src/librvnpal/src/posix/journal.c

```diff
@@ -36,15 +36,18 @@ _free_journal_handle(struct journal_handle* handle)
 }
 
 EXPORT int32_t
-rvn_open_journal_for_writes(const char *file_name, int32_t transaction_mode, int64_t initial_file_size, void **handle, int64_t *actual_size, int32_t *detailed_error_code)
+rvn_open_journal_for_writes(const char *file_name, int32_t transaction_mode, int64_t initial_file_size, int32_t durability_support, void **handle, int64_t *actual_size, int32_t *detailed_error_code)
 {
     assert(initial_file_size > 0);
 
     int32_t rc;
     
     int32_t flags = O_DSYNC | O_DIRECT;
-    if (transaction_mode == JOURNAL_MODE_DANGER)
-        flags = 0;    
+    if (durability_support == DURABILITY_NOT_SUPPORTED)
+        flags = O_DSYNC;
```

Comment:
Why? If we aren't durable, we should have no flags.

Reply:
Non Durable is different then "Danger Mode", and the reason is FS doesn't supporting O_DIRECT flag. But we do use O_DSYNC for minimum guarantee we can achieve. Otherwise we can use danger mode.   O_DIRECT is what we are missing here and it is purely the ability to ask for ack when the data is physically on the disk. 

We do want at least know the data is on the disk cache and not the OS cache with O_DSYNC..

Reply:
If we don't have `O_DIRECT` then we don't really need `O_DSYCN` either, I think. 
Same as the change you made for Windows.

Reply:
I disagree, because O_DSYNC provides a guarantee of returning from the write call only AFTER passing all the data to the disk.  
This is useful, i.e., on ungraceful reset/boot to the machine, without AC power loss. The write was done and written to the journal (but stays in the _device_ memory buffer), the system reboots and the device still have time to write is buffers physically. 

So, unless we suffer a power loss during that time to the disk, O_DSYNC can protect us from thinking we wrote something we didn't

Reply:
Actually, `O_DSYNC` force a write beyond the device buffer. In what cases are we not using `O_DIRECT`, though? What cases isn't it supported?

Reply:
NFS

Reply:
I am changing this to zero.

Reply:
Okay, I follow. Don't change it to 0 in this case, your reasoning is better.